File: C++/ejercicio4.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>

using namespace std;

void agregarContacto() {
    string nom, tel, cor;
    cin.ignore();
    cout << "Nombre completo: "; getline(cin, nom);
    cout << "Telefono: "; getline(cin, tel);
    cout << "Correo: "; getline(cin, cor);

    ofstream file("agenda.txt", ios::app);
    if (file.is_open()) {
        file << nom << "," << tel << "," << cor << endl;
        file.close();
        cout << "Contacto guardado.\n";
    }
}

void buscarContacto() {
    string buscar;
    cin.ignore();
    cout << "Ingrese el nombre a buscar: "; getline(cin, buscar);
    ifstream file("agenda.txt");
    string linea;
    bool hallado = false;

    if (file.is_open()) {
        while (getline(file, linea)) {
            stringstream ss(linea);
            string n, t, c;
            getline(ss, n, ','); getline(ss, t, ','); getline(ss, c, ',');
            
            if (n.find(buscar) != string::npos) { // Búsqueda de subcadena
                cout << "Encontrado -> Nom: " << n << " | Tel: " << t << " | Correo: " << c << "\n";
                hallado = true;
            }
        }
        file.close();
        if (!hallado) cout << "No se encontraron coincidencias.\n";
    }
}

int main() {
    int op;
    do {
        cout << "\n--- AGENDA TELEFÓNICA ---\n1. Agregar Contacto\n2. Buscar Contacto\n3. Salir\nOpción: ";
        cin >> op;
        if (op == 1) agregarContacto();
        if (op == 2) buscarContacto();
    } while (op != 3);
    return 0;
}

File: Java/ejercicio4.java
import java.io.*;
import java.util.Scanner;

public class ejercicio4 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int op;
        do {
            System.out.println("\n--- AGENDA TELEFÓNICA ---");
            System.out.println("1. Agregar Contacto\n2. Buscar Contacto\n3. Salir");
            System.out.print("Opción: "); op = sc.nextInt(); sc.nextLine();

            if (op == 1) {
                System.out.print("Nombre: "); String n = sc.nextLine();
                System.out.print("Teléfono: "); String t = sc.nextLine();
                System.out.print("Correo: "); String c = sc.nextLine();
                try (PrintWriter pw = new PrintWriter(new FileWriter("agenda.txt", true))) {
                    pw.println(n + "," + t + "," + c);
                    System.out.println("Contacto registrado.");
                } catch (IOException e) { System.out.println("Error."); }
                
            } else if (op == 2) {
                System.out.print("Nombre a buscar: "); String busq = sc.nextLine().toLowerCase();
                File f = new File("agenda.txt");
                if (!f.exists()) continue;
                try (BufferedReader br = new BufferedReader(new FileReader(f))) {
                    String l; boolean ok = false;
                    while ((l = br.readLine()) != null) {
                        String[] d = l.split(",");
                        if (d[0].toLowerCase().contains(busq)) {
                            System.out.println("-> " + d[0] + " | Tel: " + d[1] + " | Email: " + d[2]);
                            ok = true;
                        }
                    }
                    if (!ok) System.out.println("Sin coincidencias.");
                } catch (Exception e) { System.out.println("Error."); }
            }
        } while (op != 3);
    }
}
